### ISSUE TITLE
(Nx.BinaryBackend) Strided slice issue with top-dimension slice

### ISSUE THREAD

**Issue by seanmor5** [elixir-nx/nx#491]
This code in the BinaryBackend doesn't work as expected:

```elixir
iex> Nx.slice(Nx.tensor([1, 2, 3, 4, 5, 6]), [0], [6], strides: [2])
#Nx.Tensor<
  s64[3]
  [1, 2, 3]
>
```

The equivalent in NumPy:

```python
a = np.array([1, 2, 3, 4, 5, 6])
a[::2] # array([1, 3, 5])
```

This seems to only be a problem with 1-D tensors:

```elixir
iex> Nx.slice(Nx.tensor([[1, 2, 3, 4, 5, 6]]), [0, 0], [1, 6], strides: [1, 2])
#Nx.Tensor<
  s64[1][3]
  [
    [1, 3, 5]
  ]
>
```

**Comment by seanmor5** [comment]
This seems to be a problem with striding along the top dimension:

```elixir
```

**Comment by polvalente** [comment]
Looking into this

**Comment by seanmor5** [comment]
@polvalente I have a fix almost ready don't worry :)

**Comment by polvalente** [comment]
This was solved in #492
